src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -112,11 +112,14 @@ private sealed class LatestDiagnosticsForSpanGetter
                 // updating the error list simultaneously.
                 var cacheFullDocumentDiagnostics = owner.AnalyzerService.GlobalOptions.IsLspPullDiagnostics();
 
+                // Note that some callers, such as diagnostic tagger, might pass in a range equal to the entire document span.
+                // We clear out range for such cases as we are computing full document diagnostics.
+                if (range.HasValue && range.Value.Length == text.Length)
```

Comment:
i feel like this check should be `range == new TextSpan(0, text.Length);`.  I don't really understand comparing the lengths as the lengths could be the same, but referring to an overlapping region not an identical one.  the `== new TextSpan` version seems simpler and more precise as to what is being checked.